lib/recurly/payment-methods/payment-methods.js

```diff
@@ -0,0 +1,165 @@
+import Emitter from 'component-emitter';
+import Promise from 'promise';
+import recurlyError from '../errors';
+import { loadScript, loadStyle } from '../../util/dom';
+import AdyenGateway from './gateways/adyen';
+
+const GATEWAYS = {
+  adyen: AdyenGateway,
+};
+
+class PaymentMethods extends Emitter {
+  constructor (recurly, options) {
+    super();
+    this.recurly = recurly;
+    this.options = options;
+
+    this.gatewayStrategy = undefined;
+  }
+
+  async start () {
+    try {
+      await this.validateOptions();
+      const { gatewayType, paymentMethodData } = await this.getPaymentMethods();
+      this.gatewayType = gatewayType;
+      this.gatewayStrategy = await this.selectGatewayStrategy(gatewayType);
+
+      await this.setupGatewayStrategyListeners();
+      await this.loadExternalLibraries();
+      await this.initWebComponent(paymentMethodData);
+    } catch(err) {
+      this.error(err);
+    }
+  }
+
+  async submit () {
+    try {
+      const token = await this.tokenizePaymentMethod();
+      this.emit('token', token);
+    } catch (err) {
+      this.error(err);
+    }
+  }
+
+  async handleAction (paymentResponse) {
+    try {
+      this.gatewayStrategy.handleAction(paymentResponse);
+    } catch (err) {
+      this.error(err);
+    }
+  }
+
+  // private
```

Comment:
Do we have any other patterns in RJS that indicate a method is private? Or is this pretty standard?

Reply:
it [exists](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields), but we are not using it in rjs, I don't even think is supported by webpack because the feature was delivered recently.
Other common patterns are pretending `_` to the function names, but we don't use that pattern either, so I just put that comment for developer reference.